pkg/route/validation/validation.go

```diff
@@ -90,6 +113,26 @@ func validateRoute(route *routev1.Route, checkHostname bool) field.ErrorList {
 		result = append(result, err)
 	}
 
+	if route.Spec.HTTPHeaders != nil {
+		if len(route.Spec.HTTPHeaders.Actions.Response) != 0 || len(route.Spec.HTTPHeaders.Actions.Request) != 0 {
+			if route.Spec.TLS != nil && route.Spec.TLS.Termination == routev1.TLSTerminationPassthrough {
+				result = append(result, field.Invalid(field.NewPath("spec", "tls", "termination"), route.Spec.TLS.Termination, "only edge and re-encrypt routes are supported for providing customized headers."))
+			}
+		}
+		actionsPath := field.NewPath("spec", "httpHeaders", "actions")
+		if len(route.Spec.HTTPHeaders.Actions.Response) > maxResponseHeaderList {
+			result = append(result, field.Invalid(actionsPath.Child("response"), route.Spec.HTTPHeaders.Actions.Response, fmt.Sprintf("response headers list can't exceed 20 items %v", maxResponseHeaderList)))
```

Comment:
Here and also below.
```suggestion
			result = append(result, field.Invalid(actionsPath.Child("response"), route.Spec.HTTPHeaders.Actions.Response, fmt.Sprintf("response headers list can't exceed %d items", maxResponseHeaderList)))
```

Reply:
ok

Reply:
added